homeassistant/components/launch_library/sensor.py

```diff
@@ -1,69 +1,75 @@
-    DEFAULT_NAME,
+    DOMAIN,
 )
 
-_LOGGER = logging.getLogger(__name__)
 
-SCAN_INTERVAL = timedelta(hours=1)
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the sensor platform."""
 
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string}
-)
+    coordinator = hass.data[DOMAIN]
 
+    sensors = [
+        NextLaunchSensor(coordinator, "Next launch"),
+    ]
 
-async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
-    """Create the launch sensor."""
-    name = config[CONF_NAME]
-    session = async_get_clientsession(hass)
-    launches = PyLaunches(session)
+    async_add_entities(sensors, True)
 
-    async_add_entities([LaunchLibrarySensor(launches, name)], True)
 
+class LLBaseEntity(CoordinatorEntity, SensorEntity):
+    """Sensor base entity."""
 
-class LaunchLibrarySensor(SensorEntity):
-    """Representation of a launch_library Sensor."""
+    def __init__(self, coordinator: DataUpdateCoordinator, name: str) -> None:
+        """Initialize a Launch Library entity."""
+        super().__init__(coordinator)
+        self._attr_name = name
+        self._attr_unique_id = f"{DOMAIN}/{name}"
 
-    _attr_icon = "mdi:rocket"
+    def get_next_launch(self):
+        """Return next launch."""
+        return next((launch for launch in self.coordinator.data), None)
 
-    def __init__(self, api: PyLaunches, name: str) -> None:
-        """Initialize the sensor."""
-        self.api = api
-        self._attr_name = name
 
-    async def async_update(self) -> None:
-        """Get the latest data."""
-        try:
-            launches = await self.api.upcoming_launches()
-        except PyLaunchesException as exception:
-            _LOGGER.error("Error getting data, %s", exception)
-            self._attr_available = False
-        else:
-            if next_launch := next((launch for launch in launches), None):
-                self._attr_available = True
-                self._attr_native_value = next_launch.name
-                self._attr_extra_state_attributes = {
-                    ATTR_LAUNCH_TIME: next_launch.net,
-                    ATTR_AGENCY: next_launch.launch_service_provider.name,
-                    ATTR_AGENCY_COUNTRY_CODE: next_launch.pad.location.country_code,
-                    ATTR_STREAM: next_launch.webcast_live,
-                    ATTR_ATTRIBUTION: ATTRIBUTION,
-                }
+class NextLaunchSensor(LLBaseEntity):
+    """Representation of the next launch sensor."""
+
+    _attr_icon = "mdi:orbit"
+
+    @property
+    def native_value(self):
+        """Return the state of the sensor."""
+        next_launch = self.get_next_launch()
+        return next_launch.name
+
+    @property
+    def extra_state_attributes(self):
+        """Return the attributes of the sensor."""
+        next_launch = self.get_next_launch()
```

Comment:
This can return `None`, which makes the attributes below fail.

Reply:
See below question

Reply:
```suggestion
        if not (next_launch := self.get_next_launch()):
            return {}
```